protos/flyteidl/admin/execution.proto

```diff
@@ -285,6 +286,9 @@ message ExecutionSpec {
     // User setting to configure where to store offloaded data (i.e. Blobs, structured datasets, query data, etc.).
     // This should be a prefix like s3://my-bucket/my-data
     RawOutputDataConfig raw_output_data_config = 19;
+
+    // Controls how to select an available cluster on which this execution should run.
+    ClusterAssignment cluster_assignment = 20;
```

Comment:
Will we add this to the Matchable Entity to be able to specify it at the Project+Domain level? or will that happen in a separate PR?

Reply:
I was going to do in a follow-up PR but I suppose there's no harm in including it here